inc/namespace.php

```diff
@@ -176,16 +229,52 @@ function get_attachment_by_id( string $id ) :? WP_Post {
 	return $query->next_post();
 }
 
-function get_provider() : Provider {
-	static $provider = null;
+function allow_local_media() : bool {
+	$allow = defined( 'AMF_ALLOW_LOCAL_MEDIA' ) ? AMF_ALLOW_LOCAL_MEDIA : true;
+	return apply_filters( 'amf/allow_local_media', $allow );
+}
+
+function register_provider( Provider $provider ) {
+	global $amf_providers;
 
-	if ( $provider ) {
-		return $provider;
+	if ( empty( $amf_providers ) ) {
+		$amf_providers = [];
 	}
 
-	$provider = apply_filters( 'amf/provider', new BlankProvider() );
+	$amf_providers[ $provider->get_id() ] = $provider;
+}
+
+function get_providers() : array {
+	global $amf_providers;
+
+	$providers = $amf_providers ?? [];
 
-	return $provider;
+	return apply_filters( 'amf/providers', $providers );
+}
+
+function get_provider( ?string $id = null ) : Provider {
+	$providers = get_providers();
+	$provider = null;
+
+	// If no ID is passed use the first available as default.
+	if ( empty( $id ) ) {
+		$provider = array_shift( $providers );
+	} else {
+		$provider = $providers[ $id ];
+	}
+
+	// If no provider matched then error out.
+	if ( empty( $provider ) ) {
+		throw new Exception(
+			sprintf(
+				/* translators: %s: Provider class ID */
+				__( 'Media Provider with ID "%s" not found.', 'asset-manager-framework' ),
+				$id
+			)
+		);
+	}
+
+	return apply_filters( 'amf/provider', $provider, $id );
```

Comment:
Unless this is used anywhere, I'd remove it since `amf/providers` would be better suited for any changes, and would be the single point of action there.

Reply:
I removed `amf/providers` as a filter in favour of registering providers - this filter is part of the original API and is still useful for decorating a specific provider, I believe @tfrommen is using it for that purpose on a project.

In your view what makes it better suited? Are you thinking of being able to completely replace / remove / reorder providers? There are other way to achieve that eg. the `amf/script/providers` filter

Reply:
Ah, yeah, that makes sense, if we have the js filter there.